requests.exceptions.HTTPError: 404 Client Error

It seems like another error is being raised that wasn't present in 1.4.0, it isn't as obvious what is causing this 

```[2023-08-22 18:44:23 +0000] [10] [ERROR] Socket error processing request.
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/gthread.py", line 282, in handle
    keepalive = self.handle_request(req, conn)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/gthread.py", line 354, in handle_request
    util.reraise(*sys.exc_info())
  File "/usr/local/lib/python3.11/site-packages/gunicorn/util.py", line 641, in reraise
    raise value
  File "/usr/local/lib/python3.11/site-packages/gunicorn/workers/gthread.py", line 334, in handle_request
    respiter = self.wsgi(environ, resp.start_response)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2552, in __call__
    return self.wsgi_app(environ, start_response)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2532, in wsgi_app
    response = self.handle_exception(e)
               ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 2529, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1825, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1823, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask/app.py", line 1799, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/flask_oidc/views.py", line 52, in login_view
    return g._oidc_auth.authorize_redirect(redirect_uri)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/authlib/integrations/flask_client/apps.py", line 43, in authorize_redirect
    rv = self.create_authorization_url(redirect_uri, **kwargs)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/authlib/integrations/base_client/sync_app.py", line 309, in create_authorization_url
    metadata = self.load_server_metadata()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/authlib/integrations/base_client/sync_app.py", line 295, in load_server_metadata
    resp.raise_for_status()
  File "/usr/local/lib/python3.11/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://dev-530876.okta.com/oauth2/.well-known/openid-configuration


Sample of my client_secrets:

```{
  "web": {
    "client_id": "0oa3fm12fsfJWRO294x7",
    "client_secret": "[redacted]",
    "auth_uri": "https://dev-530876.okta.com/oauth2/v1/authorize",
    "token_uri": "https://dev-530876.okta.com/oauth2/v1/token",
    "issuer": "https://dev-530876.okta.com/oauth2/"
  }
}

--- a/docs/changelog.rst
+++ b/docs/changelog.rst
@@ -2,6 +2,16 @@
 Changelog
 =========
 
+UNRELEASED
+==========
+
+Deprecated
+----------
+
+- Configuration option ``OIDC_USERINFO_URL`` (and the ``userinfo_uri`` key in
+  ``client_secrets``) (:issue:`15`).
+
+
 2.0.1 (2023-08-22)
 ==================
 

--- a/flask_oidc/__init__.py
+++ b/flask_oidc/__init__.py
@@ -63,6 +63,7 @@ _CONFIG_DEPRECATED = (
     "OIDC_OPENID_REALM",
     "OVERWRITE_REDIRECT_URI",
     "OIDC_CALLBACK_ROUTE",
+    "OIDC_USERINFO_URL",
 )
 
 logger = logging.getLogger(__name__)
@@ -133,17 +134,6 @@ class OpenIDConnect:
         app.config.setdefault(
             "OIDC_CLIENT_SECRET", self.client_secrets["client_secret"]
         )
-        try:
-            app.config.setdefault(
-                "OIDC_USERINFO_URL", self.client_secrets["userinfo_uri"]
-            )
-        except KeyError:
-            if app.config.get("OIDC_USER_INFO_ENABLED", False):
-                logger.warning(
-                    'No "userinfo_uri" entry was found in the client_secrets, '
-                    "retrieving user information is disabled."
-                )
-            app.config["OIDC_USER_INFO_ENABLED"] = False
         app.config.setdefault("OIDC_USER_INFO_ENABLED", True)
         app.config.setdefault("OIDC_INTROSPECTION_AUTH_METHOD", "client_secret_post")
         app.config.setdefault("OIDC_CLOCK_SKEW", 60)

--- a/tests/client_secrets.json
+++ b/tests/client_secrets.json
@@ -7,7 +7,6 @@
         "redirect_uris": [
             "https://localhost:5000/oidc_callback"
         ],
-        "token_uri": "https://test/openidc/Token",
-        "userinfo_uri": "https://test/openidc/UserInfo"
+        "token_uri": "https://test/openidc/Token"
     }
 }

--- a/tests/test_flask_oidc.py
+++ b/tests/test_flask_oidc.py
@@ -308,27 +308,3 @@ def test_resource_server_only(client_secrets_path):
             resp = client.get(url)
             assert resp.status_code == 404
         check_token_expiry.assert_not_called()
-
-
-def test_no_userinfo_url(client_secrets, caplog):
-    app = flask.Flask("dummy")
-    client_secrets = client_secrets.copy()
-    del client_secrets["userinfo_uri"]
-    app.config["OIDC_CLIENT_SECRETS"] = {"web": client_secrets}
-    OpenIDConnect(app)
-    assert app.config["OIDC_USER_INFO_ENABLED"] is False
-    assert len(caplog.messages) == 0
-
-
-def test_no_userinfo_url_when_enabled(client_secrets, caplog):
-    app = flask.Flask("dummy")
-    client_secrets = client_secrets.copy()
-    del client_secrets["userinfo_uri"]
-    app.config["OIDC_USER_INFO_ENABLED"] = True
-    app.config["OIDC_CLIENT_SECRETS"] = {"web": client_secrets}
-    OpenIDConnect(app)
-    assert app.config["OIDC_USER_INFO_ENABLED"] is False
-    assert caplog.messages == [
-        'No "userinfo_uri" entry was found in the client_secrets, retrieving user '
-        "information is disabled."
-    ]

